### ISSUE TITLE
[Quarkus 2.13.0] Quarkus Kubernetes not compatible with Camel 3.18.x

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#4083]
See below exception from attempting to delete a pod. Likely that other APIs are also broken. We have limited test coverage to confirm.

Not sure if trying to do workarounds is viable given how many components / operations this component provides. Probably the only way to resolve things would be to align Camel 3.18 with Kubernetes Client 6.1.1.

```
2022-09-12T04:46:04.9344841Z Caused by: java.lang.NoSuchMethodError: 'java.lang.Boolean io.fabric8.kubernetes.client.dsl.PodResource.delete()'
2022-09-12T04:46:04.9345525Z 	at org.apache.camel.component.kubernetes.pods.KubernetesPodsProducer.doDeletePod(KubernetesPodsProducer.java:160)
2022-09-12T04:46:04.9346270Z 	at org.apache.camel.component.kubernetes.pods.KubernetesPodsProducer.process(KubernetesPodsProducer.java:74)
2022-09-12T04:46:04.9347036Z 	at org.apache.camel.support.AsyncProcessorConverterHelper$ProcessorToAsyncProcessorBridge.process(AsyncProcessorConverterHelper.java:66)
2022-09-12T04:46:04.9347507Z 	... 42 more
```

**Comment by oscerd** [comment]
We can't do breaking changes in 3.18.x, and moving from 5.x to 6.x will be a breaking change.

**Comment by jamesnetherton** [comment]
> moving from 5.x to 6.x will be a breaking change

Is that definitely the case? It could be breaking if folks are using the plain k8s client directly in their apps. In terms of the component functionality, everything should be the same, no (I've not checked)?

The compatibility matrix indicates the same set of k8s versions would be supported.

**Comment by oscerd** [comment]
Some of the signatures will change, so in some producer operations you'll get different classes as result. It's not really the same, and by the way we released 3.18 based on 5.x. Quarkus is just one of the runtime, we should not break existing users for one runtime

**Comment by jamesnetherton** [comment]
Quarkus 2.13.0 will stay on 5.x. 2.14.0 will have 6.x which we can align with once we upgrade to Camel 3.19.0.

https://issues.apache.org/jira/browse/CAMEL-18499
